beacon-chain/rpc/eth/beacon/blocks.go

```diff
@@ -173,6 +174,7 @@ func (bs *Server) ListBlockHeaders(ctx context.Context, req *ethpbv1.BlockHeader
 	}
 
 	isOptimistic := false
+	isFinalized := true
```

Comment:
We assume finalized and as long as we "are finalized", we update the value inside the loop.